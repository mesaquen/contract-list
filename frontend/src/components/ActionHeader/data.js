export const generateTestIds = dataTestId => ({
  component: dataTestId,
  button: `${dataTestId}-button`,
})
