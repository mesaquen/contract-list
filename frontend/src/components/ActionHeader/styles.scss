.actionHeader--container {
  display: flex;
  align-items: center;
}

.actionHeader--label-container {
  flex-grow: 1;
  align-items: center;
}

.actionHeader--subtitle {
  margin-left: 0.5rem;
}
