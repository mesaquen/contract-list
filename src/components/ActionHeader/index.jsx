import React from 'react'
import cx from 'classnames'
import Typo from '../Typography'
import Button from '../Button'
import './styles.scss'
import { generateTestIds } from './data'

const ActionHeader = ({
  className,
  title,
  subtitle,
  buttonLabel,
  onClick,
  compact,
  dataTestId,
  ...props
}) => {
  const classNames = cx('actionHeader--container', className)
  const titleSize = compact ? Typo.M : Typo.XL
  const testIds = generateTestIds(dataTestId)

  const buttonComponent =
    onClick && buttonLabel ? (
      <Button onClick={onClick} raised data-testid={testIds.button}>
        {buttonLabel}
      </Button>
    ) : null

  return (
    <div className={classNames} {...props} data-testid={testIds.component}>
      <div className="actionHeader--label-container">
        <Typo size={titleSize} bold>
          {title}
        </Typo>
        {subtitle && (
          <Typo size={Typo.S} className="actionHeader--subtitle">
            {subtitle}
          </Typo>
        )}
      </div>
      {buttonComponent}
    </div>
  )
}

export default ActionHeader
