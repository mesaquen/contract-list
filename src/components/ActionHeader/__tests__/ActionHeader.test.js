import { cleanup, render, fireEvent, screen } from '@testing-library/react'
import ActionHeader from '../index'

let callback

beforeEach(() => {
  cleanup()
  callback = jest.fn()
})

const renderComponent = (props = {}) => {
  return (
    <ActionHeader
      title="Title"
      buttonLabel="Action"
      dataTestId="action-header"
      {...props}
    />
  )
}

describe('ActionHeader', () => {
  it('renders correctly with no action', () => {
    const { asFragment } = render(renderComponent())
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly with action', () => {
    const { asFragment } = render(renderComponent({ onClick: callback }))
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly on compact layout', () => {
    const { asFragment } = render(renderComponent({ compact: true }))
    expect(asFragment()).toMatchSnapshot()
  })

  it('fires callback when clicked', async () => {
    render(renderComponent({ onClick: callback }))
    const element = await screen.findByTestId('action-header-button')
    fireEvent.click(element)
    expect(callback).toBeCalledTimes(1)
  })
})
