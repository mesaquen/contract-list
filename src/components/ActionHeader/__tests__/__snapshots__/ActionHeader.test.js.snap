// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionHeader renders correctly on compact layout 1`] = `
<DocumentFragment>
  <div
    class="actionHeader--container"
    data-testid="action-header"
  >
    <div
      class="actionHeader--label-container"
    >
      <span
        class="typo--m typo--bold typo--text-primary"
      >
        Title
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ActionHeader renders correctly with action 1`] = `
<DocumentFragment>
  <div
    class="actionHeader--container"
    data-testid="action-header"
  >
    <div
      class="actionHeader--label-container"
    >
      <span
        class="typo--xl typo--bold typo--text-primary"
      >
        Title
      </span>
    </div>
    <button
      class="button raised"
      data-testid="action-header-button"
    >
      Action
    </button>
  </div>
</DocumentFragment>
`;

exports[`ActionHeader renders correctly with no action 1`] = `
<DocumentFragment>
  <div
    class="actionHeader--container"
    data-testid="action-header"
  >
    <div
      class="actionHeader--label-container"
    >
      <span
        class="typo--xl typo--bold typo--text-primary"
      >
        Title
      </span>
    </div>
  </div>
</DocumentFragment>
`;
